### ISSUE TITLE
docker compose for windows containers cannot parse volume mapping correctly

### ISSUE THREAD

**Issue by HuanhuanSunMSFT** [docker/compose#4012]
Here is a very simple docker compose file.
```
version: '2'

services:
  testproject:
    image: testproject
    build:
      context: .
      dockerfile: Dockerfile
    volumes: 
      - .:C:\app
    ports:
      - "80"
networks:
  default:
    external:
      name: nat
```
Here is the error I get when running compose up:
ERROR: for fulldotnet  Cannot create container for service fulldotnet: invalid volume spec "/c/app": invalid volume specification: '\c\app'

docker-compose version 
```
docker-compose version 1.8.1, build 004ddae
docker-py version: 1.10.3
CPython version: 2.7.12
OpenSSL version: OpenSSL 1.0.2h  3 May 2016
```

**Comment by pennywisdom** [comment]
Also getting similar on 1.9.0dev:

```batch
docker-compose version 1.9.0dev, build eac01a7
docker-py version: 1.10.3
CPython version: 2.7.12
OpenSSL version: OpenSSL 1.0.2h  3 May 2016
```

Have tried numerous variations to see if anything will work.

Running Windows Server 2016

```powershell
Platform ServicePack Version      VersionString
-------- ----------- -------      -------------
 Win32NT             10.0.14393.0 Microsoft Windows NT 10.0.14393.0
```

**Comment by shin-** [comment]
I'm pretty sure this is #3923 .

**Comment by friism** [comment]
@HuanhuanSunMSFT There's a build available now that doesn't do the windows path rewriting: https://ci.appveyor.com/project/docker/compose/build/master-2420/artifacts

Recommend closing this issue since it's most likely a replicate of #3923

**Comment by HuanhuanSunMSFT** [comment]
Thanks. @shin- @friism I just tried the latest compose but I got this error:
```
PS C:\Users\husun\Desktop\aspnet> docker-compose up
WARNING: The COMPOSE_CONVERT_WIN32_PATHS variable is not set. Defaulting to a blank string.
Traceback (most recent call last):
  File "<string>", line 3, in <module>
  File "compose\cli\main.py", line 62, in main
  File "compose\cli\main.py", line 114, in perform_command
  File "compose\cli\main.py", line 835, in up
  File "compose\project.py", line 372, in up
  File "compose\project.py", line 539, in warn_for_swarm_mode
  File "site-packages\docker\api\daemon.py", line 33, in info
  File "site-packages\docker\utils\decorators.py", line 47, in inner
  File "site-packages\docker\client.py", line 139, in _get
  File "site-packages\requests\sessions.py", line 477, in get
  File "site-packages\requests\sessions.py", line 465, in request
  File "site-packages\requests\sessions.py", line 573, in send
  File "site-packages\requests\adapters.py", line 370, in send
  File "site-packages\requests\packages\urllib3\connectionpool.py", line 544, in urlopen
  File "site-packages\requests\packages\urllib3\connectionpool.py", line 374, in _make_request
  File "httplib.py", line 1133, in getresponse
  File "httplib.py", line 390, in __init__
  File "site-packages\docker\transport\npipesocket.py", line 99, in makefile
ValueError: buffer size must be strictly positive
docker-compose returned -1
```

Here is the compose file:
```
version: '2'

services:
  dotnetfull:
    image: dotnetfull
    build:
      context: .
      dockerfile: Dockerfile
    ports:
      - "8000"
    volumes:
      - ./publishoutput/:c:\app
networks:
  default:
    external:
      name: nat
```

docker compose version:
```
PS C:\Users\husun\Desktop\aspnet> docker-compose version
docker-compose version 1.9.0dev, build 31904b7
docker-py version: 1.10.3
CPython version: 2.7.12
OpenSSL version: OpenSSL 1.0.2h  3 May 2016
```

docker version:
```
PS C:\Users\husun\Desktop\aspnet> docker version
Client:
 Version:      1.13.0-dev
 API version:  1.25
 Go version:   go1.7.1
 Git commit:   e097c2a
 Built:        Tue Oct  4 19:42:36 2016
 OS/Arch:      windows/amd64

Server:
 Version:      1.13.0-dev
 API version:  1.25
 Go version:   go1.7.1
 Git commit:   e097c2a
 Built:        Tue Oct  4 19:42:36 2016
 OS/Arch:      windows/amd64
```

**Comment by HuanhuanSunMSFT** [comment]
I tried set host to `$env:DOCKER_HOST="127.0.0.1:2375"` and it eliminates the error.
https://github.com/docker/compose/issues/3901

**Comment by shin-** [comment]
@HuanhuanSunMSFT Thank you for testing this out. Looks like we still have some issues with named pipes support.

**Comment by HuanhuanSunMSFT** [comment]
@shin-  Is there any workaround available so that I can test the piece of the volume mapping

**Comment by friism** [comment]
@HuanhuanSunMSFT `$Env:DOCKER_HOST = "localhost:2375"`

**Comment by HuanhuanSunMSFT** [comment]
$Env:DOCKER_HOST = "localhost:2375" fixes this issue though

Close this one since it's partially resolved and also a duplicate of docker/compose#3923
